### PR TITLE
chore(ci): use sccache for benchmarks

### PR DESCRIPTION
**Pull request by tobz**
This PR introduces the usage of `sccache` for our benchmark steps.  `sccache` caches compiled dependencies and reuses them, which should allow our benchmark steps to avoid unnecessary recompilation, not only between the `master`/`current compilations within a single run, but also between different PRs and so on.

Signed-off-by: Toby Lawrence <toby@nuclearfurnace.com>

### CODE REVIEW

File: .github/workflows/benches.yml [hunk 1]
```diff
@@ -67,6 +67,18 @@ jobs:
...
       - run: echo "::add-matcher::.github/matchers/rust.json"
+      # Install and configure sccache, including telling Cargo to use it.
+      - uses: actions-rs/install@v0.1
+        with:
+          crate: sccache
+          version: latest
+      - name: Start sccache
+        env:
+          SCCACHE_REDIS: ${{ secrets.SCCACHE_REDIS }}
+          SCCACHE_IDLE_TIMEOUT: 0
```
Comment (jszwedko) [inline .github/workflows/benches.yml:78]: Just noting for other readers that this disables the timeout.

**Review by jszwedko — APPROVED** [review]
👍 nice, I'm interested to see the speed-up from this. Do you expect we'll also add this to the other workflows?

**Comment by tobz** [comment]
I do expect we will, yes.  Right now, I'm just trying to measure the speedup and have a lower-stakes part of the CI workflow to ensure it works properly.

File: .github/workflows/benches.yml (hunk 1 above)
Comment (spencergilbert) [reply .github/workflows/benches.yml:79]: Worth adding as a comment?

**Pull request merged**